docs/testing.md

```diff
@@ -32,27 +32,80 @@ Those can be ignored and `/skip`ed. In example:
 ```
 Upstream issue: https://github.com/openshift/ci-operator-prowgen/issues/79 
 
+## Verify 
+
+This repository runs multiple checks to **verify** if it complies with standard go conventions.  The scripts from the target run through a series of checks to ensure the repository remains compliant with standard go practices. This can be ran **locally** with the command `make verify`.  When creating a pull-request, this tests is part of the acceptance criteria. 
+
+```makefile
+verify:
+    ./hack/verify/validate-generated.sh
+    go vet ./...
+    ./hack/verify/validate-code-format.sh
+    ./hack/verify/validate-util.sh
+    ./hack/verify/validate-codecov.sh
+    go run ./hack/validate-imports/validate-imports.go   cmd hack pkg test
+    ./hack/verify/validate-sec.sh
+```
+
 ## Unit test
 
 Packages under `pkg` directory contain their individual unit tests.
-To run all unit tests locally, execute: `make unit` or `go test ./...`
+
+To run all unit tests **locally**, execute: `make unit`
+
+```makefile
+unit: generate testinsights
+  go test ./... -coverprofile=coverage.out -covermode=atomic -json | ./testinsights
+```
+_Note: This will **generate** bindata.go files when using this Makefile target as well as testinsights._
+
+or 
+
+at the root of the repository `openshift-azure/` 
+
+```go 
+go test ./... -v 
+```
+
 To run a single package's tests:
-`go test ./pkg/util/tls`
+```go
+go test ./pkg/util/tls
+```
 
 To run a subset of tests in a package:
-`go test -run TestFoo ./pkg/util/tls`
+```go
+go test -run TestFoo ./pkg/util/tls
```

Comment:
does anyone run this on the cli? - yikes I hope not, it is so much easier to run it within your editor..